tests/std/tests/P0896R4_P1614R2_comparisons/test.cpp

```diff
@@ -293,19 +293,7 @@ STATIC_ASSERT(test_compare_three_way<compares_as<strong_ordering>, compares_as<s
 
 // Validate dynamic properties of compare_three_way, ranges::equal_to, ranges::not_equal_to, ranges::less,
 // ranges::less_equal, ranges::greater, ranges::greater_equal
-template <class T, class U, class R>
-constexpr void assert_three_way(T const& t, U const& u, R const result) { // TRANSITION, VSO-905257
-#if !defined(__clang__) && !defined(__EDG__)
-    // <=> expressions that resolve to builtin operators are incorrectly lvalues of const-qualified type on MSVC, so
-    // they are rejected by std::three_way_comparable and std::three_way_comparable_with.
-    if constexpr (!std::is_class_v<std::remove_cvref_t<T>> && !std::is_class_v<std::remove_cvref_t<U>>) {
-        return;
-    } else
-#endif // !defined(__clang__) && !defined(__EDG__)
-    {
-        assert(compare_three_way{}(t, u) == result);
-    }
-}
+#define assert_three_way(t, u, result) assert((compare_three_way{}(t, u) == result))

```

Comment:
Minor note: the preprocessor's insect-level intelligence is capable of matching parentheses (although not angle brackets or braces), so the double parentheses in this `assert` aren't necessary. However, strict preprocessor cleanliness would involve parenthesizing `t`, `u`, and `result`:
```suggestion
#define assert_three_way(t, u, result) assert(compare_three_way{}((t), (u)) == (result))
```
As this is test code, I view this as optional.

Reply:
I went ahead and pushed this change, mostly because being strictly clean in test code provides better precedent for people studying that code and then writing product code. (I think it matters more for obscure stuff like the preprocessor than simple stuff like `using`-directives where convenience outweighs the downsides of creating precedent for laziness.)

Thanks for cleaning up these compiler workarounds - we really appreciate it!